tests/Feature/GoogleDriveConfigApiControllerTest.php

```
@@ -511,4 +512,384 @@ public function test_sync_cannot_trigger_for_other_users_config(): void
+        $response->assertStatus(403)
+            ->assertJson(['error' => ['code' => 'PERMISSION_DENIED']]);
+    }
+
+    public function test_folders_returns_parent_id_from_query_param(): void
+    {
+        $config = GoogleDriveConfig::factory()->create(['user_id' => $this->user->id]);
+
+        $capturedParentId = null;
+        $mock = $this->createMock(GoogleDriveService::class);
+        $mock->method('listFolders')
+            ->willReturnCallback(function ($cfg, $parentId) use (&$capturedParentId) {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$cfg'. (undefined)

(UnusedFormalParameter)